lib/bbc-microbit-io.js

```diff
@@ -137,21 +137,27 @@ BBCMicrobitIO.prototype.digitalRead = function(pin, handler) {
 BBCMicrobitIO.prototype._onDiscover = function(microbit) {
   debug('on discover', microbit);
 
-  this._microbit = microbit;
+  //Certain Intel BT Chipsets seem to require a scan stop before connections are allowed, otherwise they'll throw a: Connection Rejected due to Limited Resources 
+  BBCMicrobit.stopScanning(() => {
+    this._microbit = microbit;
 
-  this._microbit.on('pinDataChange', this._onPinDataChange.bind(this));
+    this._microbit.on('pinDataChange', this._onPinDataChange.bind(this));
 
-  this._microbit.once('disconnect', function() {
-    this.emit('disconnect');
-  }.bind(this));
-
-  this._microbit.connectAndSetUp(function() {
-    this.emit('connect');
+    this._microbit.once('disconnect', function() {
+      this.emit('disconnect');
+    }.bind(this));
 
-    this._microbit.subscribePinData(function() {
-      this.emit('ready');
+    this._microbit.connectAndSetUp(function(error) {
+      if (error) {
+        console.error(error);
```

Comment:
Whichever suits your style really :) I had a "throw" initially but felt that was too strong, but thought an emit might be "too weak". Can change.